### ISSUE TITLE
OOM always!

### ISSUE THREAD

**Issue by sinAshish** [CVLAB-Unibo/inr2vec#2]
Hello, 

I am following your code for shape generation and applying to my dataset. 
I have already created a dataset in the same format used in your experiments, however, while training INR2vec, I always get OOM during [ validation step](https://github.com/CVLAB-Unibo/inr2vec/blob/56f8b40c93d0e900c3158921e995c08c53050b54/pcd_completion/train_inr2vec.py#L231). My RAM and GPU memory keep on increasing during this stage, while during the training iteration it is well below 40% (12 G Titan X), it seems like there is a memory leak or something I am not able to figure out.
I  tried reducing the num of query samples, network and embedding size, but still no help, it gets OOM for each sample during validation step.

Could you please help!

**Comment by lykius** [comment]
Hi @sinAshish,

One possibility is that the network is not trained enough when reaching the validation step and thus the function `sample_pcds_from_udfs` keeps accumulating points without being able of reconstructing the surface.
One quick test that you could do is delaying the validation. In our case, the dataset was huge so running the validation after the first epoch was not a problem, but if you are training on a small dataset one epoch could be not enough.

Let me know.

**Comment by sinAshish** [comment]
@lykius yes my dataset is quite small, training on a bunch of 100 objects. I tried to increase the interval but that didnt help either. so I ended up changing the code to sample N points randomly and use those to estimate the gradients, something like:
```python
samples.requires_grad = True
samples_pred = udfs_func(samples, indices)
samples_grad = self.gradient(samples, samples_pred)
grad_directions = F.normalize(samples_grad, dim=-1)
pred_pcd = samples - grad_directions * samples_pred.unsqueeze(-1)
```
my shapes are quite complex, so might need to train longer, cause the shape space looks awfully similar

**Comment by sinAshish** [comment]
And also , for the shape generation, it's too much of a hassle to setup `latent_points_3d` for training, I had to change alot for the code to work in tensorflow2, if you can want I can shoot a PR for the patch

**Comment by lykius** [comment]
That would be amazing!

**Comment by lykius** [comment]
Anyway there is no need to update the code in `latent_points_3d` repository to work in tensorflow2. We used the original environment from `latent_points_3d`, thus with tensorflow1. The script that we provide should work out of the box with such environment.
